pkg/config/config.go

```diff
@@ -43,107 +60,174 @@ type ApplicationInfo struct {
 	ActiveComponent string `json:"activeComponent"`
 }
 
-// Config records odo's configuration
-type Config struct {
-	// odo specific configuration settings
-	OdoSettings OdoSettings `json:"settings"`
+// GlobalConfig stores all the config related to odo, its the superset of
+// local config.
+type GlobalConfig struct {
 	// remember active applications and components per project
 	// when project or applications is switched we can go back to last active app/component
 
 	// Currently active application
 	// multiple applications can be active but each one has to be in different project
 	// there shouldn't be more active applications in one project
 	ActiveApplications []ApplicationInfo `json:"activeApplications"`
+
+	// Odo settings holds the odo specific global settings
+	OdoSettings *OdoSettings `json:"settings,omitempty"`
+}
+
+// LocalConfig holds all the config relavent to a specific Component.
+type LocalConfig struct {
+	ComponentSettings *ComponentSettings `json:"componentSettings,omitempty"`
 }
 
-// ConfigInfo records information about odo's configuration
-type ConfigInfo struct {
-	Filename string
-	Config
+// GlobalConfigInfo wraps the global config and provides helpers to
+// serialize it.
+type GlobalConfigInfo struct {
+	Filename     string `json:"fileName,omitempty" yaml:"fileName,omitempty"`
```

Comment:
why we have `yaml` tag here, but not anywhere else?
`yaml` marshall looks for `json` tags, we shouldn't this info twice.

Reply:
ack

Reply:
resolved

Reply:
resolved